app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -175,13 +176,47 @@ export const convertFlatRecordsToNestedRecords = (records: RQAPI.Record[]) => {
       const parentNode = recordsMap[record.collectionId] as RQAPI.CollectionRecord;
       if (parentNode) {
         parentNode.data.children.push(recordState);
+        const getBreadCrumbOptions = recordsMap[parentNode.id].breadcrumbOptions || [];
+        recordState.breadcrumbOptions = [
+          ...getBreadCrumbOptions,
+          { id: parentNode.id, name: parentNode.name, type: parentNode.type },
+        ];
       }
     } else {
       updatedRecords.push(recordState);
     }
   });
 
-  return updatedRecords;
+  return { updatedRecords, recordsMap };
 };
 
 export const getEmptyPair = (): KeyValuePair => ({ id: Math.random(), key: "", value: "", isEnabled: true });
+
+export const getBreadCrumbOptions = (
+  record: RQAPI.Record,
+  prefixOption: {
+    pathname: string;
+    label: string;
+    disabled?: boolean;
+    isEditable?: boolean;
+  }
+) => {
+  let formattedOptions = [prefixOption];
+  const options = record?.breadcrumbOptions || [];
+  formattedOptions = [
+    ...formattedOptions,
+    ...options.map((option) => ({
+      pathname: `${prefixOption.pathname}/${option.type}/${option.id}`,
+      label: option.name,
+      disabled: option.type === RQAPI.RecordType.COLLECTION,
+    })),
+  ];
+
+  formattedOptions.push({
```

Comment:
I don't think breadcrumbOptions needs to be pushed to db. They should be calculated at runtime

Reply:
they are not pushed to db they are getting calculated only one time while rendering the sidebar UI then it is stored in apiClient context.